content/blogposts/2022/code-intelligence-platform.md

```diff
@@ -0,0 +1,38 @@
+---
+title: "From code search toward a code intelligence platform"
+publishDate: 2022-09-27T10:00-07:00
+description: Finding things in code is good, but learning things about code is great. Sourcegraph's code intelligence platform is built to help.
+tags: [blog]
+slug: "code-search-to-code-intelligence"
+published: false
+heroImage: 
+socialImage: 
+---
+
+No matter how good your other dev tools, coding is still complex. Even if you've configured your editor just right, sped up your test suite, set up continuous deployment, added observability, etc., coding is still complex. Higher-level languages, libraries, frameworks, cloud providers, etc., that encapsulate complexity just let you spend your complexity budget elsewhere. The overall complexity of code has increased, not declined, over time, and there's no sign of that trend changing.
+
+As consumers of software, we (mostly) benefit from this phenomenon in the form of better technology in our everyday lives.
+
+But as developers of software, we all feel the pain of code complexity: the struggle to understand big codebases, not being able to get in flow, encountering tech debt that turns a tiny fix into a multi-day ordeal, all the wasted time, etc. 
+
+On top of this, devs are more important to their companies than ever. Devs have gone from being viewed as cost centers to being seen as the center of product development, so the issue of complexity is a business problem just as much as a dev problem.
+
+Code search is helpful in tackling complexity, but it's not enough. Code search is for finding things *in* code (matching lines in files). What helps you learn things *about* code? Google went from helping you find things *on* the web (matching web pages), to helping you learn things *from* the web (information and answers), and we're doing the same with code.
+
+This is what we call a code intelligence platform, and compared to code search, it's more helpful, more times per day. **A code intelligence platform helps devs learn and understand information about their code so they can operate more effectively, with a deeper understanding of their codebase and how it works.** 
+
+A code intelligence platform should persist data beyond the code itself and make that data accessible. Here are some examples of what we see the platform doing in the future:
```

Comment:
What does "persist data beyond the code itself" mean?

Reply:
we could rephrase as "store data beyond the code itself" or even "compute and store data beyond the code itself" (I assume as used here it means we don't just have your code, we have code insights trackers, and batch change burndown charts, and future dependency graphs and ownership signals, and CI job outputs, and test coverage, and so on – all data that we created and/or save rather than pull from other systems, like the code host)

Reply:
```suggestion
A code intelligence platform should compute and store data beyond the code itself and make that data accessible. Here are some examples of what we see the platform doing in the future:
```